### PR TITLE
Add tail sampling blog post

### PR DESCRIPTION
**Pull request by reese-lee**
This is the vendor-neutral version of my blog post on tail sampling with OpenTelemetry. A version of this post was originally published on the New Relic blog. For this post, I've removed all mentions to New Relic so it's completely vendor-neutral.

### CODE REVIEW

File: content/en/blog/2022/tail-sampling/index.md
Comment (svrnm) [inline content/en/blog/2022/tail-sampling/index.md:5]: you can put `*(New Relic)*` behind your name if you like.

**Review by svrnm — APPROVED** [review]
awesome, great read, lgtm

File: content/en/blog/2022/tail-sampling/index.md
Comment (cartermp) [inline content/en/blog/2022/tail-sampling/index.md:137]: I know this wasn't in the NR blog post, but a major thing to consider as well is that otel doesn't propagate metadata that can let a backend re-weight counts (e.g., total events, P95, P99, etc.) - and so you're getting a look at sampled data, but not accurate measurements w.r.t. _all_ data. Sort of a middle ground between sampling as-possible with OTel today and not sampling at all.

Is that something that you'd want to describe in this post as well? I'm fine with it not being here, it's just something that came to mind that isn't mentioned.

**Review by cartermp — APPROVED** [review]

File: content/en/blog/2022/tail-sampling/index.md
Comment (svrnm) [reply content/en/blog/2022/tail-sampling/index.md:137]: Interesting point, how much would probabilistic sampling help with that?

Comment (cartermp) [reply content/en/blog/2022/tail-sampling/index.md:137]: Unless there's a way to pass a long the effective sample rate, then I can't see how that could help get accurate counts on sampled data, no.

Comment (reese-lee) [reply content/en/blog/2022/tail-sampling/index.md:137]: Ah, thank you for this! I'm interested in including it, is there anything to expand on further with this point?

Comment (cartermp) [reply content/en/blog/2022/tail-sampling/index.md:137]: Yeah, so basically the gist of it is this premise:

> In the monitoring industry there’s a complicated and frustrating conversation that persisted over the years: how do you deal with the enormous volume of data generated by instrumentation? On one side of the aisle, you will find a cohort of vendors and developers telling you that you have to sample data, followed immediately by another group telling you that sampling will ruin the accuracy of incident analysis. They’re both right.

From https://blog.sentry.io/2022/08/23/introducing-dynamic-sampling

If you're only taking 1/5 traces, then unless there's something the backend can use to understand that the data it's operating on is only 1/5 the size of the _real_ data, any measurement (e.g., counting total spans) will be inaccurate. And so if you're working with statistically significant data, then that _might_ be fine for some scenarios, but could be bad for others where you actually do need an accurate accounting of data.

One middle ground here is to have metadata attached to spans/events/etc. that tells a backend what the sample rate is.

Using that example of sampling 1/5 traces, if there's metadata in those sampled traces that says something to the effect of "I'm one out of five!", then a backend can multiple values by 5 to actually be accurate for measures like "total events during this period of time". This is one of the core concepts behind the current sampling SIG that'd getting hammered out, and it's understandably not rolled out in any of the SDK or collector sampling capabilities because of that.

Comment (reese-lee) [reply content/en/blog/2022/tail-sampling/index.md:137]: @cartermp 

Thank you! I added a paragraph to the last section. Can you let me know if I've summed it up well, or if I should modify it?

>Another limitation to consider is that since OpenTelemetry does not propagate
metadata that would let a backend re-weight counts – for example, P95, P99, and
total events – you’re only getting a look at the measurements of sampled data,
not accurate measurements in regards to _all_ data. Let’s say you’ve configured
your sampler to keep 25% of all traces. If the backend doesn’t know it’s only
operating on 25% of all data, any measurement it produces will be inaccurate.
One way to get around this is to attach metadata to your spans that tells the
backend what the sample rate is, that will allow the backend to then accurately
measure things such as total span count for a given period of time. The Sampling
SIG is currently working on this concept.

**Review by cartermp — APPROVED** [review]

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
@cartermp , @reese-lee : looks good to me, let's wait for approval from @open-telemetry/blog-approvers and we're good.

**Review by austinlparker — APPROVED** [review]

**Comment by svrnm** [comment]
@reese-lee , as discussed at the comms meeting I want to give every blog post a few days to stay on top, I just merged #1721, so I'll wait a few days (maybe until Monday?) to merge this. hope that's OK!

**Comment by svrnm** [comment]
We are live: https://opentelemetry.io/blog/2022/tail-sampling/ 🎉

**Pull request merged**
